src/main/java/com/ably/kafka/connect/client/DefaultAblyBatchClient.java

```diff
@@ -113,20 +123,38 @@ private boolean isError(final HttpPaginatedResponse response) {
      */
     private void handlePartialFailure(
         HttpPaginatedResponse response,
-        MessageGroup messageGroup) throws AblyException {
-
-        do {
-            JsonElement[] batchResponses = response.items();
-            for (JsonElement batchResponse : batchResponses) {
-                for (AblyChannelPublishException error : getFailedChannels(batchResponse)) {
-                    logger.debug("Submission to channel {} failed", error.channelName, error);
-                    final List<SinkRecord> failedRecords = messageGroup.recordsForChannel(error.channelName);
-                    sendMessagesToDlq(failedRecords, error);
-                }
-            }
+        List<RecordMessagePair> recordMessagePairs) throws AblyException {
+        Iterator<RecordMessagePair> recordMessagePairItr = recordMessagePairs.iterator();
+        Iterator<JsonElement> itemsItr = Arrays.stream(response.items()).iterator();
+
+        while (recordMessagePairItr.hasNext() && itemsItr.hasNext()) {
+            RecordMessagePair recordMessagePair = recordMessagePairItr.next();
+            JsonObject batchSpecResponse = itemsItr.next().getAsJsonObject();
+
+            int failureCount = batchSpecResponse.
+                getAsJsonPrimitive("failureCount").getAsInt();
+
+            if (failureCount == 0) continue;
 
-            response = response.next();
-        } while (response != null);
+            JsonArray results = batchSpecResponse.getAsJsonObject().getAsJsonArray("results");
+
+            if (results == null || results.size() != 1)
+                throw AblyException.fromErrorInfo(new ErrorInfo("Inconsistent batch response: result field should contain single element", 500));
```

Comment:
Do we need an Ably error code here? We have 500, but should probably have 50000 too?